src/main/java/games/battlelore/components/MapTile.java

```diff
@@ -0,0 +1,168 @@
+package games.battlelore.components;
+
+import core.components.Component;
+import utilities.Utils;
+import java.util.ArrayList;
+
+public class MapTile extends Component {
+
+    protected int locationX, locationY;
+    private ArrayList<Unit> units;
+    private Terrain terrain;
+
+    public enum TileArea {
+        NA, left, mid, right;
+    }
+
+    public int getLocationX() {
+        return locationX;
+    }
+
+    public int getLocationY() {
+        return locationY;
+    }
+
+    public MapTile(int locationX, int locationY, ArrayList<Unit> units, Terrain terrain, int componentID) {
```

Comment:
This type of constructor (with component ID) should not be used for things other than making copies. Component IDs are set automatically for all components in the game, so that each component has a unique ID; should not expose this externally.